clab/clab.go

```diff
@@ -83,16 +81,48 @@ func NewContainerLab(opts ...ClabOption) *CLab {
 		Config:   new(Config),
 		TopoFile: new(TopoFile),
 		m:        new(sync.RWMutex),
-		Nodes:    make(map[string]*Node),
+		Nodes:    make(map[string]*types.Node),
 		Links:    make(map[int]*Link),
 	}
+
+	err := c.initMgmtNetwork()
+	if err != nil {
+		log.Fatalf("initMgmtNetwork: %s", err)
+	}
+
 	for _, o := range opts {
 		o(c)
 	}
+
 	return c
 }
 
-func (c *CLab) CreateNode(ctx context.Context, node *Node, certs *Certificates) error {
+// initMgmtNetwork sets management network config
+func (c *CLab) initMgmtNetwork() error {
+	if c.Config.Mgmt.Network == "" {
+		c.Config.Mgmt.Network = dockerNetName
+	}
+	if c.Config.Mgmt.IPv4Subnet == "" && c.Config.Mgmt.IPv6Subnet == "" {
+		if c.Config.Mgmt.IPv4Subnet == "" {
+			c.Config.Mgmt.IPv4Subnet = dockerNetIPv4Addr
+		}
+		if c.Config.Mgmt.IPv6Subnet == "" {
+			c.Config.Mgmt.IPv6Subnet = dockerNetIPv6Addr
+		}
+	}
+	// init docker network mtu
+	if c.Config.Mgmt.MTU == "" {
+		m, err := getDefaultDockerMTU()
```

Comment:
likely this func needs to be runtime-scoped? i.e. with containerd/podman the logic should be runtime-specific as `docker0` interface won't be there

Reply:
so should we move this whole function under the runtime interface? or do you want to just skip MTU here and set it in the runtime?

Reply:
tbh, I don't fully understand the need to initMgmtNetwork. why can't we just defer to the runtime to set it all up (after all it's managing it's own IPAM). maybe you can elaborate @hellt ?

Reply:
I think we can start with a small change of adding a new runtime-specific function named `DefaultMgmtNetMTU` which will return the default management network MTU using the runtime-specific methods.

Reply:
@networkop the `initMgmtNetwork` was created to set the default configuration values for mgmtNetName (clab) and default subnets for ipv4/6

It might be that this can be removed/changed with the default values to be passed over to the respective runtime func that creates the network

Reply:
right, this makes sense. but this `getDefaultDockerMTU()` creates circular dependency between the runtime and the init function: you're building values to configure runtime while at the same time relying on runtime to provide some of these values.
In particular, let's assume we move this behind a runtime interface, we would then rely on runtime being set up before we can call initMgmt function, which looks like a scope creep -- we'd need to modify all the config parsing tests to init a runtime in order to set the MTU value.

Reply:
then maybe has an util function that takes the runtime name and returns the default MTU for a given runtime value?
i.e. for docker it will use the existing func that gets the mtu of a `docker0` iface

Reply:
yeah, that might work. let me see

Reply:
actually, it will have the same problem, i.e. you need to know the runtime name when you populate your default values.

Reply:
at this stage, i think the only right solution (I think) is to leave MTU unchanged and do a check (if empty, then lookup) inside the `CreateNet` function, which is the only place where MTU is used.

Reply:
but the runtime name is always provided via `rt` value, and has a `docker` default, no?

Reply:
yes, there's always a runtime set. The problem is that runtime is currently optional and most of the `config_test.go` functions are set up just `WithTopoFile`. It's not a problem adding a `WithRuntime` to them as well, it'll work, but it is an indication of leaking dependencies -- your config test code should not really rely on a runtime being set.
But if you're ok with this, I'm happy to add runtime to every test in `config_test.go`, so they'll all look like 👍 
```diff
	for name, tc := range tests {
		t.Run(name, func(t *testing.T) {
			opts := []ClabOption{
				WithTopoFile(tc.got),
+                              WithRuntime(...),
			}
```

Reply:
maybe we should defer this till we sort the other things

Reply:
another issue with getting default MTU is determining which bridge to use. Some runtimes may defer to CNI plugins to setup the networking, which means we'd have to parse all CNI configuration files in order to extract the name of the bridge, and it may not even exist in case of a P2P CNI plugin. So maybe we should just have the magic `getDefaultMTU` function that looks picks the lowest MTU off several well-known bridge names, and, if none found, falls back to some safe default, e.g. 1400. wdyt @hellt